src/index.ts

```diff
@@ -752,3 +753,41 @@ app.on('will-finish-launching', () => {
     });
   });
 });
+
+app.on(
+  'certificate-error',
+  (event, _webContents, _url, _error, certificate, callback) => {
+    // On certificate error we disable default behaviour (stop loading the page)
+    // and we then say "it is all fine - true" to the callback
+    event.preventDefault();
+
+    const useSelfSignedCertificates =
+      retrieveSettingValue(
+        'useSelfSignedCertificates',
+        DEFAULT_APP_SETTINGS.useSelfSignedCertificates,
+      ) === true;
+
+    // Check if the certificate is trusted
+    if (!useSelfSignedCertificates) {
+      callback(false);
+      return;
+    }
+
+    const trustedCerts = readCerts();
+    if (!trustedCerts) {
+      callback(false);
+      return;
+    }
```

Comment:
There is some repetition here that you might be able to chain together into one if condition.

Reply:
I want to prevent the execution of `readCerts()` if the setting is disabled on the UI - this is because otherwise the function will open and read all files there every time the event fires, which is a waste of resources... Maybe there is a better way of doing this @mcmxcdev ? Any suggestion?

Reply:
Ah makes sense, early return then. All good!

Reply:
you can still do this, correct?
```suggestion
    // Check if the certificate is trusted
    if (!useSelfSignedCertificates || !readCerts()) {
      callback(false);
      return;
    }
```

Reply:
Yes, but I need the result returned by readCerts() if it is true to be saved 🙈

Maybe if you get the time to refactor it better I would appreciate it... I'm really not seeing how to 🙈